### ISSUE TITLE
IDC: Return the expected URLs when blocking sync

### ISSUE THREAD

**Issue by ebinnion** [Automattic/jetpack#5453]
Our mocks now call for showing the current site URL as well as the expected URL on WPCOM. We need to find a way to return these values when we return the error object.

So, perhaps what we do is return an array that contains an error property? Maybe something like:

``` php
array(
    'error'         => WP_Error(),
    'wpcom_siteurl' => $siteurl,
    'wpcom_home'    => $home,
)
```

Then [around here](https://github.com/Automattic/jetpack/blob/master/sync/class.jetpack-sync-actions.php#L148-L158), we could maybe add a check to be something like:

``` php
if ( ! $result ) {  
    return $error;
}

if ( is_array( $result ) && isset( $result['error'] ) ) {
    $error = $result['error'];

    if ( is_wp_error( $error ) && 'jetpack_url_mismatch' === $error->get_error_code() ) {
        Jetpack_Options::update_option(
            'sync_error_idc',
            Jetpack::get_sync_error_idc_option(),
        );
    }
}
```

Then, in `get_sync_error_idc_option()` we could change what we actually store.

**Comment by ebinnion** [comment]
@dereksmart has already done some work in #5367, before we got updated designs. So, we'll want to keep an eye on that.

**Comment by dereksmart** [comment]
Solved by #5414
